diff -rupN avbin-20091022svn.old/src/avbin.c avbin-20091022svn/src/avbin.c
--- avbin-20091022svn.old/src/avbin.c	2009-10-22 23:15:06.000000000 -0400
+++ avbin-20091022svn/src/avbin.c	2009-10-23 00:45:33.000000000 -0400
@@ -321,9 +321,13 @@ int avbin_decode_audio(AVbinStream *stre
     if (stream->type != CODEC_TYPE_AUDIO)
         return AVBIN_RESULT_ERROR;
 
-    used = avcodec_decode_audio2(stream->codec_context, 
+    AVPacket pkt;
+    av_init_packet(&pkt);
+    pkt.data = data_in;
+    pkt.size = size_in;
+    used = avcodec_decode_audio3(stream->codec_context, 
                                  (int16_t *) data_out, size_out,
-                                 data_in, size_in);
+                                 &pkt);
 
     if (used < 0)
         return AVBIN_RESULT_ERROR;
@@ -344,9 +348,13 @@ int avbin_decode_video(AVbinStream *stre
     if (stream->type != CODEC_TYPE_VIDEO)
         return AVBIN_RESULT_ERROR;
 
-    used = avcodec_decode_video(stream->codec_context, 
+    AVPacket pkt;
+    av_init_packet(&pkt);
+    pkt.data = data_in;
+    pkt.size = size_in;
+    used = avcodec_decode_video2(stream->codec_context, 
                                 stream->frame, &got_picture,
-                                data_in, size_in);
+                                &pkt);
     if (!got_picture)
         return AVBIN_RESULT_ERROR;
 
