diff -rupN avbin-20091022svn.old/src/avbin.c avbin-20091022svn/src/avbin.c
--- avbin-20091022svn.old/src/avbin.c	2009-10-22 23:15:06.000000000 -0400
+++ avbin-20091022svn/src/avbin.c	2009-10-22 23:24:07.000000000 -0400
@@ -55,7 +55,7 @@ static void avbin_log_callback(void *ptr
     static char message[8192];
     const char *module = NULL;
 
-    if (level > av_log_level || !user_log_callback)
+    if (level > av_log_get_level() || !user_log_callback)
         return;
 
     if (ptr)
@@ -99,7 +99,7 @@ AVbinResult avbin_init()
 
 AVbinResult avbin_set_log_level(AVbinLogLevel level)
 {
-    av_log_level = level;
+    av_log_set_level(level);
     return AVBIN_RESULT_OK;
 }
 
